File: python/lld/vending-machine/idle_state.py
from state import State
from has_money_state import HasMoneyState

class IdleState(State):
    """Waiting for the first coin."""

    def name(self) -> str: return "IDLE"

    def insert_coin(self, m, cents):
        m.balance += cents
        m.set_state(HasMoneyState())

    # select_product, dispense, refund — inherited _reject handles them

File: java/lld/vending-machine/IdleState.java
// Waiting for the first coin. Only inserting money does anything here.
public class IdleState extends State {
    public String name() { return "IDLE"; }

    public void insertCoin(VendingMachine m, int cents) {
        m.balance += cents;
        m.setState(new HasMoneyState());
    }

    // selectProduct, dispense, refund — not overridden, inherited reject handles them
}